src/Consolonia.Core/Drawing/StreamGeometryImpl.cs

```diff
@@ -0,0 +1,197 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Avalonia;
+using Avalonia.Media;
+using Avalonia.Platform;
+
+namespace Consolonia.Core.Drawing
+{
+    internal class StreamGeometryImpl : IStreamGeometryImpl
+    {
+        private readonly List<Rectangle> _fills;
+        private readonly List<Line> _strokes;
+        private Rect _bounds;
+
+        public StreamGeometryImpl()
+        {
+            _strokes = new List<Line>();
+            _fills = new List<Rectangle>();
+        }
+
+        public IReadOnlyList<Line> Strokes => _strokes;
+
+        public IReadOnlyList<Rectangle> Fills => _fills;
+        // private SKPath _path;
+
+        public Rect Bounds => _bounds;
+
+        public double ContourLength => _strokes.Sum(l => l.ContourLength);
+
+
+        public IStreamGeometryImpl Clone()
+        {
+            var clone = new StreamGeometryImpl();
+            foreach (Line line in _strokes) clone._strokes.Add(line);
+            foreach (Rectangle rect in _fills) clone._fills.Add(rect);
+            clone._bounds = _bounds;
+            return clone;
```

Comment:
Half-deep cloning?) I'm not aware how it's used, but is it fine we keep same instances of lines and rectangles for the clone? Should we do serializing/deserializing or other deep-cloning techniques?

Reply:
I updated this to clone the classes, all of the parts that "copied" now are structs and immutable.